identity-wallet/src/state/profile_settings/reducers/update_sorting_preference.rs

```diff
@@ -58,45 +63,401 @@ pub async fn sort_credentials(state: AppState, _action: Action) -> Result<AppSta
     let mut credentials: Vec<DisplayCredential> = state.credentials.clone();
     let preferences: Preferences<CredentialSortMethod> = state.profile_settings.sorting_preferences.credentials.clone();
 
-    let name_az = |a: &DisplayCredential, b: &DisplayCredential| a.display_name.cmp(&b.display_name);
-    let issuance_new_old =
-        |a: &DisplayCredential, b: &DisplayCredential| a.metadata.date_issued.cmp(&b.metadata.date_issued);
-    let added_new_old =
-        |a: &DisplayCredential, b: &DisplayCredential| a.metadata.date_added.cmp(&b.metadata.date_added);
+    let display_names: Vec<String> = credentials
+        .iter()
+        .map(|credential| credential.display_name.clone())
+        .collect();
+    let sorted_display_names = sort(display_names, state.profile_settings.locale.clone());
 
-    credentials.sort_by(match preferences.sort_method {
-        CredentialSortMethod::NameAZ => name_az,
-        CredentialSortMethod::IssueDateNewOld => issuance_new_old,
-        CredentialSortMethod::AddedDateNewOld => added_new_old,
-    });
+    let name_az = |a: &DisplayCredential, b: &DisplayCredential| {
+        let cmp_a = sorted_display_names
+            .iter()
+            .position(|display_name| display_name == &a.display_name)
+            .unwrap();
+        let cmp_b = sorted_display_names
+            .iter()
+            .position(|display_name| display_name == &b.display_name)
+            .unwrap();
+        cmp_a.cmp(&cmp_b)
+    };
+
+    match preferences.sort_method {
+        CredentialSortMethod::NameAZ => credentials.sort_by(name_az),
+        CredentialSortMethod::IssueDateNewOld => credentials.sort_by(|a: &DisplayCredential, b: &DisplayCredential| {
+            a.metadata.date_issued.cmp(&b.metadata.date_issued).reverse()
+        }),
+        CredentialSortMethod::AddedDateNewOld => credentials.sort_by(|a: &DisplayCredential, b: &DisplayCredential| {
+            a.metadata.date_added.cmp(&b.metadata.date_added).reverse()
+        }),
+    };
 
     if preferences.reverse {
         credentials.reverse();
     }
 
+    // In this block we check if there are credentials without an issue-date.
+    // When the issue-date is empty, we sort these credentials alphabetically and add them to the bottom.
+    if preferences.sort_method == CredentialSortMethod::IssueDateNewOld {
+        let mut credentials_empty_issue_date: Vec<DisplayCredential> = credentials
+            .iter()
+            .filter(|credential| credential.metadata.date_issued.is_empty())
+            .cloned()
```

Comment:
This could potentially be quite an expensive operation because it will clone all `DisplayCredential`s without a `date_issued`. This data type can be rather large especially because of its `data` field. 

I would suggest to came up with a solution here that doesn't clone any `DisplayCredential` e.g.:
```rust
    if preferences.sort_method == CredentialSortMethod::IssueDateNewOld {
        let (mut credentials_with_date_issued, mut credentials_without_date_issued) = credentials
            .into_iter()
            // splits the `credentials` into to vectors.
            .partition::<Vec<_>, _>(|credential| !credential.metadata.date_issued.is_empty());

        credentials_without_date_issued.sort_by(name_az);
        credentials_with_date_issued.append(&mut credentials_without_date_issued);
       // assign the value reconstructed credentials back to `credentials`.
        credentials = credentials_with_date_issued
    };
```